src/compiler/compiler.cc

```diff
@@ -1515,9 +1516,17 @@ toit::Program* construct_program(ir::Program* ir_program,
 
   ASSERT(_sorted_by_inheritance(ir_program->classes()));
 
-  if (run_optimizations) optimize(ir_program, propagated_types);
+  if (run_optimizations) optimize(ir_program, oracle);
   tree_shake(ir_program);
 
+  if (propagated_types) {
```

Comment:
Needs a comment on how ordering is preserved between the two runs.

Reply:
Done.